go-controller/pkg/ovn/controller/services/node_tracker.go

```diff
@@ -76,11 +78,13 @@ func (ni *nodeInfo) l3gatewayAddressesStr() []string {
 	return out
 }
 
-func newNodeTracker(zone string, resyncFn func(nodes []nodeInfo)) *nodeTracker {
+func newNodeTracker(zone string, resyncFn func(nodes []nodeInfo), netInfo util.NetInfo) *nodeTracker {
```

Comment:
do we need this really per network?
this tracker can just be global and used in each network right?

Reply:
also the chassis template changes to SBDB per node; => if we do this per network; its going to be duplicated right?

Reply:
There are parts of the node tracker that are relative to the network itself, like `gatewayRouterName` and `switchName`, which are used all throughout the code by the services controller to add LBs to the DB; there are other parts which are more relative to the node itself (`hostAddresses`, `l3gatewayAddresses`).
I agree that this can be further optimized, maybe in a follow up PR?